### PR TITLE
replace `from_config` everything with `from_archive`

### PR DESCRIPTION
**Pull request by joelgrus**
this is how `evaluate` and `predict` should work, and it's one step closer to serving actual trained models in the demo

* `evaluate` command now takes an archive file rather than config + weights
* `predict` command now takes an archive file rather than config
* `load_archive` now returns a tuple (Model, Vocabulary, Config)
* replace `Predictor.from_config` with `.from_archive`
* instantiate all predictors in server + tests with archives 
* add test fixtures for model archives

### CODE REVIEW

File: allennlp/models/archival.py [hunk 1]
```diff
@@ -64,13 +69,16 @@ def load_model(archive_file: str, cuda_device: int = -1) -> Model:
     # Load config
     config = Params.from_file(os.path.join(tempdir, _CONFIG_NAME))
 
-    # Instantiate model
-    model = Model.load(config,
+    # Load vocabulary
+    vocab = Vocabulary.from_files(os.path.join(tempdir, "vocabulary"))
```
Comment (matt-gardner) [inline allennlp/models/archival.py]: Minor optimization: this is going to load the vocab from files twice.  Might as well just pull out `model._vocab`, which also does this.

Comment (joelgrus) [inline allennlp/models/archival.py]: originally I was going to do this. here's the problem. `model._vocab` is just a convention, we have no way of enforcing it. so what I was going to do was give `Model` a constructor that takes `vocab` and make every subclass constructor call it. then `model.vocab` would be guaranteed to make sense. I'm still happy to do that, but this seemed simpler (and more inefficient). thoughts?

Comment (matt-gardner) [inline allennlp/models/archival.py]: Hmm, good point.  I think it's probably worth adding the vocab to the model constructor.  @DeNeutoy?

Comment (joelgrus) [inline allennlp/models/archival.py]: I decided to go ahead and do that, as it allowed me to remove `vocabulary` from the `Archive` namedtuple as redudant

**Pull request merged**